plugins/modules/ec2_instance.py

```diff
@@ -1332,20 +1333,48 @@ def value_wrapper(v):
     param_mappings = [
         ParamMapper('ebs_optimized', 'EbsOptimized', 'ebsOptimized', value_wrapper),
         ParamMapper('termination_protection', 'DisableApiTermination', 'disableApiTermination', value_wrapper),
+        ParamMapper('security_groups', 'Groups', 'groupSet', value_wrapper),
+        ParamMapper('security_group', 'Groups', 'groupSet', value_wrapper),
         # user data is an immutable property
         # ParamMapper('user_data', 'UserData', 'userData', value_wrapper),
     ]
 
     for mapping in param_mappings:
         if params.get(mapping.param_key) is not None and mapping.instance_key not in skip:
             value = AWSRetry.jittered_backoff()(ec2.describe_instance_attribute)(Attribute=mapping.attribute_name, InstanceId=id_)
-            if params.get(mapping.param_key) is not None and value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
-                arguments = dict(
-                    InstanceId=instance['InstanceId'],
-                    # Attribute=mapping.attribute_name,
+            if mapping.param_key not in ['security_groups', 'security_group']:
+                if params.get(mapping.param_key) is not None and value[mapping.instance_key]['Value'] != params.get(mapping.param_key):
+                    arguments = dict(
+                        InstanceId=instance['InstanceId'],
+                        # Attribute=mapping.attribute_name,
+                    )
+                    arguments[mapping.instance_key] = mapping.add_value(params.get(mapping.param_key))
+                    changes_to_apply.append(arguments)
+            elif bool(params.get(mapping.param_key)) is not False:
+                if params.get('vpc_subnet_id'):
+                    subnet_id = params.get('vpc_subnet_id')
+                else:
+                    default_vpc = get_default_vpc(ec2)
+                    if default_vpc is None:
+                        raise module.fail_json(
```

Comment:
If you're calling fail_json you don't need to 'raise'

```suggestion
                        module.fail_json(
```